src/components/engagement/engagement.service.ts

```diff
@@ -202,6 +244,13 @@ export class EngagementService {
       throw new BadRequestException('That Project type is not Translation');
     }
 
+    if (await this.getLEByProjectAndLanguage(projectId, languageId)) {
+      throw new DuplicateException(
+        'engagement.languageId',
+        'LanguageEngagement with this projectId and languageId already exists'
```

Comment:
Would you mind tweaking the wording?
```suggestion
        'Engagement for this project and language already exists'
```